pkg/render/monitor/monitor.go

```diff
@@ -109,15 +109,20 @@ func (mc *monitorComponent) Objects() ([]client.Object, []client.Object) {
 		render.CreateNamespace(common.TigeraPrometheusNamespace, mc.cfg.Installation.KubernetesProvider),
 	}
 
+	// Create role/role bindings and service account first.
+	// Operator needs the create/update roles for Alertmanger configuration secret for example.
+	toCreate = append(toCreate,
+		mc.prometheusServiceAccount(),
```

Comment:
I don't think prometheusServiceAccount needs to move does it? The role and binding aren't for that account.